disnake/welcome_screen.py

```diff
@@ -0,0 +1,173 @@
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional
+
+from .partial_emoji import PartialEmoji
+from .utils import MISSING, _get_as_snowflake
+
+if TYPE_CHECKING:
+    from .guild import Guild
+    from .state import ConnectionState
+    from .types.welcome_screen import (
+        WelcomeScreen as WelcomeScreenPayload,
+        WelcomeScreenChannel as WelcomeScreenChannelPayload,
+    )
+__all__ = ("WelcomeScreen", "WelcomeScreenChannel")
+
+
+class WelcomeScreenChannel:
+    """Represents a Discord welcome screen channel.
+
+    .. versionadded:: 2.4
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The ID of the guild channel this welcome screen channel represents.
+    description: :class:`str`
+        The description of this channel in the official UI.
+    emoji: Optional[:class:`PartialEmoji`]
+        The emoji associated with this channel's welcome message, if any.
+    """
+
+    def __init__(self, *, id: int, description: str, emoji: Optional[PartialEmoji] = None):
+        self.id = id
+        self.description = description
+        self.emoji = emoji
+
+    def __repr__(self):
+        return f"<WelcomeScreenChannel id={self.id!r} emoji={self.emoji!r} description={self.description!r}>"
+
+    @classmethod
+    def _from_data(
+        cls, *, data: WelcomeScreenChannelPayload, state: ConnectionState
+    ) -> WelcomeScreenChannel:
+        emoji_id = _get_as_snowflake(data, "emoji_id")
+        emoji_name = data.get("emoji_name") or None
+        emoji = None
+        if emoji_name:
+            emoji = PartialEmoji.with_state(state, name=emoji_name, id=emoji_id)
+
+        return cls(id=int(data["channel_id"]), description=data["description"], emoji=emoji)
+
+    def to_dict(self):
+
+        result = {
+            "channel_id": self.id,
+            "description": self.description,
+        }
+        if self.emoji is not None:
+            result["emoji_id"] = self.emoji.id
+            result["emoji_name"] = self.emoji.name
+
+        return result
+
+
+class WelcomeScreen:
+    """Represents a Discord welcome screen for a :class:`Guild`.
+
+    .. versionadded:: 2.4
+
+    Attributes
+    ----------
+    description: :class:`str`
+        The welcome screen description.
```

Comment:
According to the documentation, this is nullable, correct me if I'm wrong please.
![image](https://user-images.githubusercontent.com/67214928/153600261-34ed4ff5-c347-44bc-ab64-90b91906dded.png)

Reply:
https://github.com/DisnakeDev/disnake/pull/339/commits/cf6aded359eaa078cfdcaa5997022dcd2fa8287b